### ISSUE TITLE
Failure to query collection folders

### ISSUE THREAD

**Issue by ederlf** [guruhq/py-sdk#70]
Hello, 

Since the recent change from boards to folders in our collections, I have been facing this issue when trying to get the folders from a collection using the sdk. I am using the latest version `1.1.0`.

Attempting: (I have tried with both the collection name, the id on the URL and the id returned by the API. All had the same effect)
```
folders = guru_client.get_folders(collection="My collection that totally exists")
```

will raise:
```
Traceback (most recent call last):
  File "/home/ederlf/.local/lib/python3.10/site-packages/requests/models.py", line 910, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ederlf/.local/lib/python3.10/site-packages/guru/core.py", line 2304, in get_folders
    collection_obj = self.get_collection(collection, cache=True)
  File "/home/ederlf/.local/lib/python3.10/site-packages/guru/core.py", line 442, in get_collection
    return find_by_name_or_id(self.get_collections(cache), collection)
  File "/home/ederlf/.local/lib/python3.10/site-packages/guru/core.py", line 464, in get_collections
    return [Collection(c, guru=self) for c in response.json()]
  File "/home/ederlf/.local/lib/python3.10/site-packages/requests/models.py", line 917, in json
    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
requests.exceptions.JSONDecodeError: [Errno Expecting value] : 0
```

I did some debugging to check what was the returned response and it seems I am hitting 401 (Unauthorized). 

```
> /home/ederlf/.local/lib/python3.10/site-packages/guru/core.py(464)get_collections()
-> return [Collection(c, guru=self) for c in response.json()]
(Pdb) response.json()
*** requests.exceptions.JSONDecodeError: [Errno Expecting value] : 0
(Pdb) response
<Response [401]>
(Pdb) 
```

Could you help me figure out if the problem is on my side? 
With my guru token I used to be able to list board without issues. I can also query for folders directly via the API without issues. Plus, I am the collection owner and the folders have inherited permissions from it.

**Comment by ederlf** [comment]
I have also tried to use. It's also hitting 401.

```
folder = guru_client.get_folder("Folder", "Collection")
```

One interesting thing I've noticed when debugging it. 
This is the url get_folder seems to be looking for
`'https://api.getguru.com/api/v1/folders/Folder`. 

If I try to use that url path with the API, I get a 404.
To have that working with the API, I had to use the id of the folder that shows up on Guru's URL of the folder:
``'https://api.getguru.com/api/v1/folders/<Folder id>`

Anyway, the issue seems to be on the basic authentication. I have just wondered if I get past the authentication issue, if that url could be also an issue to make the correct request.
